From 1f1dc3fc4c2e5e2d94e86dfc7235a4b762da2e72 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@qt.io>
Date: Fri, 16 Nov 2018 16:01:30 +0100
Subject: [PATCH] Fix gamma-correction in QCoreTextFontEngine with Mojave

The code was previously assuming font-smoothing was only used with
A32 font antialiasing, so the corresponding gamma-correction was not
performed.

Task-number: QTBUG-71075
Task-number: QTBUG-71946
Change-Id: I68d8304cf18638239d8bfac32c67333f16ccc7bd
Reviewed-by: Lars Knoll <lars.knoll@qt.io>
---
 src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm b/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm
index 271f07e0c17..7fb22c06756 100644
--- a/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm
+++ b/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm
@@ -736,8 +736,7 @@ bool QCoreTextFontEngine::shouldSmoothFont() const
 
 bool QCoreTextFontEngine::expectsGammaCorrectedBlending() const
 {
-    // Only works well when font-smoothing is enabled
-    return (glyphFormat == Format_A32) && !(fontDef.styleStrategy & (QFont::NoAntialias | QFont::NoSubpixelAntialias));
+    return shouldSmoothFont();
 }
 
 QImage QCoreTextFontEngine::imageForGlyph(glyph_t glyph, QFixed subPixelPosition, const QTransform &matrix)
-- 
2.16.3

